### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by liach**
Moved to a RailcraftBlockProperties class
Add BlockMachineAlpha (Not tested)
Improved ITileRotate

### CODE REVIEW

File: src/main/java/mods/railcraft/common/blocks/RailcraftBlockProperties.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+ * ******************************************************************************
+ *  Copyright 2011-2016 CovertJaguar
+ *
+ *  This work (the API) is licensed under the "MIT" License, see LICENSE.md for details.
+ * ***************************************************************************
```
Comment (liach) [inline src/main/java/mods/railcraft/common/blocks/RailcraftBlockProperties.java:6]: Oh I didn't realize the header.

File: src/main/java/mods/railcraft/common/items/firestone/ItemFirestoneRefined.java [hunk 2]
```diff
@@ -147,6 +147,9 @@ public void addInformation(ItemStack stack, EntityPlayer player, List<String> in
 
     @Override
     public EnumActionResult onItemUse(ItemStack stack, EntityPlayer player, World world, BlockPos pos, EnumHand hand, EnumFacing side, float hitX, float hitY, float hitZ) {
+        if (world.isRemote) {
```
Comment (liach) [inline src/main/java/mods/railcraft/common/items/firestone/ItemFirestoneRefined.java:150]: This is intended to fix that reported crash,

File: src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java [hunk 3]
```diff
@@ -154,13 +156,25 @@ public int getCookTime() {
...
         if (masterOven != null)
             return masterOven.facing;
         return facing;
     }
 
+    @Override
+    public void setFacing(@Nonnull EnumFacing facing) {
+        TileSteamOven masterOven = (TileSteamOven) getMasterBlock();
+        if (masterOven != null) {
+            masterOven.facing = facing;
```
Comment (liach) [inline src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java:171]: Dunno how to set the facing of each block...

Comment (CovertJaguar) [reply src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java:171]: I suspect they just ask the master

File: src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java [hunk 4]
```diff
@@ -154,13 +156,25 @@ public int getCookTime() {
...
+        }
+        this.facing = facing;
+        WorldPlugin.setBlockState(worldObj, getPos(), getBlockState().withProperty(BlockMachineAlpha.FRONT, facing));
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java]: This is certainly wrong, the meta can't support more than 16 states

**Review by CovertJaguar — CHANGES_REQUESTED** [review]

File: src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java (hunk 4 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java]: But it does nothing with the meta. It just update the facing of the tile being called.

Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java]: For now I will remove this setting of block states.

Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/machine/alpha/TileSteamOven.java]: Fixed in next commit.

**Comment by liach** [comment]
Wait, do not merge yet, I got some issue with manipulator textures.

**Comment by liach** [comment]
I've fixed for those comments but somehow these fixed places does not show as outdated diffs.

File: src/main/java/mods/railcraft/common/items/firestone/ItemFirestoneRefined.java (hunk 2 above)
Comment (CovertJaguar) [reply src/main/java/mods/railcraft/common/items/firestone/ItemFirestoneRefined.java:150]: In the future, put stuff like this in a separate PR.

**Comment by liach** [comment]
@CovertJaguar Anything else to fix?

**Review by CovertJaguar — APPROVED** [review]

**Comment by CovertJaguar** [comment]
Nothing particularly outstanding, but I'm going to sit on this until I start working on this stuff.

**Comment by Ristellise** [comment]
keep it open so that CJ will remember tp fix it later.

**Comment by liach** [comment]
Well those fixes are really minor. They are not that important.

**Comment by CovertJaguar** [comment]
Most important thing is this commit is the Firestone fix...which I know I'm going to forget.

**Pull request closed without merging**